File: server/src/main/java/me/rolgalan/marvelw/server/model/ServerCollectionList.java
package me.rolgalan.marvelw.server.model;

import java.util.List;

import javax.annotation.Nullable;

/**
 * Created by Roldán Galán on 20/11/2017.
 */

public class ServerCollectionList {

    int available;
    String collectionURI;
    List<CollectionItem> items;

    public int getAvailable() {

        return available;
    }

    public String getCollectionURI() {

        return collectionURI;
    }

    public List<CollectionItem> getItems() {

        return items;
    }

    public static class CollectionItem {

        String resourceURI;
        String name;
        @Nullable
        String type;

        public String getResourceURI() {

            return resourceURI;
        }

        public String getName() {

            return name;
        }

        @Nullable
        public String getType() {

            return type;
        }
    }
}

File: server/src/main/java/me/rolgalan/marvelw/server/model/ServerCollectionList.kt
package me.rolgalan.marvelw.server.model

/**
 * Created by Roldán Galán on 20/11/2017.
 */

class ServerCollectionList {

    var available: Int = 0
        internal set
    var collectionURI: String = ""
        internal set
    var items: List<CollectionItem>? = null
        internal set

    class CollectionItem {

        var resourceURI: String = ""
            internal set
        var name: String = ""
            internal set
        var type: String? = null
            internal set
    }
}
